libraries/networking/src/udt/SendQueue.cpp

```diff
@@ -24,7 +25,27 @@
 #include "Socket.h"
 
 using namespace udt;
-using namespace std::chrono;
+
+// This class is not thread-safe
+class DoubleLock {
+public:
+    DoubleLock(std::mutex& mutex1, std::mutex& mutex2) : _mutex1(mutex1), _mutex2(mutex2) { };
+    ~DoubleLock() { unlock(); }
+    
+    DoubleLock(const DoubleLock&) = delete;
+    DoubleLock& operator=(const DoubleLock&) = delete;
+    
+    bool locked() { return _locked; }
+    
+    bool try_lock() { return _locked = (std::try_lock(_mutex1, _mutex2) == -1); }
+    void lock() { std::lock(_mutex1, _mutex2); _locked = true; }
+    void unlock() { if (_locked) { _mutex1.unlock(); _mutex2.unlock(); _locked = false; } }
+    
+private:
+    std::atomic<bool> _locked { false };
+    std::mutex& _mutex1;
+    std::mutex& _mutex2;
+};
```

Comment:
Yes, I used the implementation you posted yesterday.
Since you hadn't pushed anything with it inside, I assumed you wanted me to add it.